### PR TITLE
Explicit Browser Support

### LINKED ISSUE #1477: better error display for unsupported browsers
We see occasional users attempt to use cellxgene with unsupported browsers (eg, IE on Win 7).  It would be nice if the front-end could present the user with something other than a hard failure.

### LINKED ISSUE #1479: List supported browsers in README
See #1477

### PR DESCRIPTION
**Pull request by seve**
This PR adds transparency and explicit support for our supported browsers.

This PR includes the following changes:
* `README.md`
  * Add section on supported browsers
* `client/package.json`
  * addition of `browserslist` attribute
  * dependencies and dev-dependencies additons
* `client/configuration/babel/babel.dev.js` + `client/configuration/babel/babel.prod.js`
  * switch from `modern-browsers` to `@babel/preset-env` for browerslists support on polyfill
* EDIT(8/3): `client/configuration/webpack/webpack.config.dev.js` + `client/configuration/webpack/webpack.config.prod.js` + `client/configuration/webpack/webpack.config.shared.js`
  * Add`script-ext-html-webpack-plugin` for async bundle loading support
  * Add polyfills to entry
  * Expand babel transpiling to node_modules and `.jsx` files to guarantee browser support
  * Add `obsolete-webpack-plugin` to display message on supported browsers
  * EDIT(8/3): Created shared config to consolidate shared aspects of webpack config in a single config
* EDIT(8/3): `client/configuration/webpack/obsoleteHTMLTemplate.html`
  *  Created HTML file to hold obsolete html template
* `client/configuration/eslint/eslint.js`
  * Extend compat plugin to ensure no unsupported API usage
  * List included polyfills

 
--- 
Closes #1477, Closes #1479

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,33 +14,48 @@ Whether you need to visualize one thousand cells or one million, cellxgene helps
...
+
+cellxgene currently supports the following browsers:
+
+- Google Chrome 61+
+- Edge 15+
+- Firefox 60+
+- Safari 10.1+
+
+Please [file an issue](https://github.com/chanzuckerberg/cellxgene/issues/new/choose) if you find that your browser is not supported or if would like us to expand our support to other browsers.
+
```
Comment (seve) [inline README.md:58]: There are a number of browsers that we don't yet know if we support (eg. Opera, QQ, Navigator)

Comment (seve) [inline README.md:58]: Do the readme additions need to be reflected somewhere on the cellxgene site? If so, where?

File: README.md [hunk 2]
```diff
@@ -14,33 +14,48 @@ Whether you need to visualize one thousand cells or one million, cellxgene helps
...
+### Supported browsers
+
+cellxgene currently supports the following browsers:
+
+- Google Chrome 61+
+- Edge 15+
+- Firefox 60+
+- Safari 10.1+
+
+Please [file an issue](https://github.com/chanzuckerberg/cellxgene/issues/new/choose) if you find that your browser is not supported or if would like us to expand our support to other browsers.
```
Comment (ambrosejcarr) [inline README.md:57]:
```suggestion
Please [file an issue](https://github.com/chanzuckerberg/cellxgene/issues/new/choose) if you would like us to add support for an unsupported browser.
```

A bit more concise.

File: README.md (hunk 1 above)
Comment (seve) [reply README.md]: cc: @sidneymbell

File: client/configuration/webpack/webpack.config.dev.js [hunk 3]
```diff
@@ -73,6 +80,44 @@ module.exports = {
       inject: true,
       template: path.resolve("index.html"),
     }),
+    new ObsoleteWebpackPlugin({
+      name: "obsolete",
```
Comment (tihuan) [inline client/configuration/webpack/webpack.config.dev.js:84]: tiny nit:

I see that the template is used in both `config.dev.js` and `config.prod.js`

In the spirit of DRY and making the template easier to maintain/update the template down the road, can we try to extract the template as an html file and use Webpack's [HTML Loader](https://webpack.js.org/loaders/html-loader/) to load the template as string and include it here? (I've never done this before, but looks possible lol)

E.g.,

1. Move the template into a file `unsupported-browser.html`
2. Import the file in this config file

```ts
const unsupportedBrowserTemplate = require('unsupported-browser.html')

new ObsoleteWebpackPlugin({
  ...
  template: unsupportedBrowserTemplate,
  ...
})
```
3. We also need to add the html-loader plugin to our config files too:

```ts
module.exports = {
  module: {
    rules: [
      {
        test: /\.html$/i,
        loader: 'html-loader',
      },
    ],
  },
};
```
---

Lastly, this is probably a separate ticket, but I'm thinking given that dev and prod config share a few common settings, maybe we can extract the common ones out, and import the common part into dev and prod config files and use [webpack-merge](https://github.com/survivejs/webpack-merge)?

E.g., Meta example:

1. Common: https://github.com/chanzuckerberg/khepri/blob/develop/webpack/webpack.common-client.js
2. Dev: https://github.com/chanzuckerberg/khepri/blob/develop/webpack/webpack.client-dev.js
3. Prod: https://github.com/chanzuckerberg/khepri/blob/develop/webpack/webpack.client-prod.js

File: client/configuration/webpack/webpack.config.prod.js [hunk 4]
```diff
@@ -89,6 +96,44 @@ module.exports = {
       decodeEntities: false,
       minify: false,
     }),
+    new ObsoleteWebpackPlugin({
+      name: "obsolete",
+      template:
+        "<script>" +
```
Comment (colinmegill) [inline client/configuration/webpack/webpack.config.prod.js:102]: Since this is a string, could it be in a different file, pulled into webpack?

**Pull request merged**